lib/metrics/context.js

```diff
@@ -156,6 +159,18 @@ module.exports = function (log, config) {
   }
 
   /**
+   * Attempt to clear previously-stashed metrics context metadata.
+   *
+   * @name clearMetricsContext
+   * @this request
+   */
+  function clear () {
+    return P.resolve()
+      .then(() => getMemcached().delAsync(getKey(getToken(this))))
+      .catch(() => {})
```

Comment:
This is the clumsy part.

When the flow completion event is `account.login`, the call to `getToken` fails because the request object doesn't have a uid and token id. That's fine because we only really need this function for the `account.signed` case, where we do have the data we need.

So I'm ignoring the error. But maybe I should switch it round so that we only call `clear` when we really need to? Fwiw, I didn't like how that turned out because it introduced an ugly conditional in `log.flowEvent`.

And the error thrown from `getToken` is legitimately thrown, because we want to log occurrences of it if they happen inside `gather`.